libyul/optimiser/ExpressionSimplifier.cpp

```diff
@@ -63,6 +71,33 @@ void ExpressionSimplifier::visit(Expression& _expression)
 					)
 						startArgument = Literal{debugDataOf(startArgument), LiteralKind::Number, "0"_yulstring, {}};
 				}
+
+			// Replace add(X, A) with sub(X, -A) if it would save gas.
```

Comment:
This is seemed like a good place for this, because `ExpressionSimplifier` is where `sub -> add` replacements mentioned in the issue are done.

And this is outside of main `SimplificationRules` loop, because it has to be done after all simplifications, and also to be able to estimate gas savings.